## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


# 1.Introduction

# 1.1 Overview

The giant tech company Microsoft has decided that its time to venture into creating original video content and
therefore want to creat a movie studio
For this project, I will use exploratory data analysis to generate insights to  guide them on the way forward


# 1.2 Business understanding

Microsoft sees all the big companies creating original video content and they want to get in on the fun. 
They have decided to create a new movie studio, but they don’t know anything about creating movies. 
I have  been charged with exploring what types of films are currently doing the best at the box office. 
and give three recommendation

For me to translate these findings into actionable insights in order to help 
<br>Microsoft build a  new movie studio I should ask the important business questions <br>
<br>     1.Which type of films are the highest grossing in the box office?<br>
<br>      2.What are the highest rated movies in the box office?<br>
 <br>    3.What are the genres of the highest grosssing and highest rated movies in the box office?<br>

Completing data analysis and answering the above questions will solve Microsost's quest for the Movie industry. 
They will have more information on the type of movies they are suppose to air in there anticipated Movie industry

# 2  Data understanding

The data I am going to use in this Data analysis is  from Various websites contain information about the Box office
These datasets will enable me to understand the trends in the movie industry and therfore I will be able to produce the relevant 
information for microsoft so that they will be able to correctly excecute they dream of becoming succesful in the movie industry

These  datasets include<br>
 **movie_gross.csv**<br>
 **movie_info.tsv**<br>
 **rt.reviews.tsv**<br>
 **tmdb.movies.csv**<br>
 **movie_budgets.csv**<br>

are in the zippedData folder are from the following websites<br>
[Box Office Mojo](https://www.boxofficemojo.com/)<br>
[IMDB](https://www.imdb.com/)<br>
[Rotten Tomatoes](https://www.rottentomatoes.com/)<br>
[TheMovieDB](https://www.themoviedb.org/)<br>
[The Numbers](https://www.the-numbers.com/)<br>

# 2.1 Loading Libraries and Datasets

In [75]:
    # importing the packages I will be using for this project
    import zippedData
    import numpy as np
    import sqlite3
    import pandas as pd
    conn = sqlite3.connect("data/im.db")
    import csv
    import matplotlib.pyplot as plt
    %matplotlib inline
    import seaborn as sns

In [76]:
#loading datasets
df_movie_gross = pd.read_csv('data/bom.movie_gross.csv')
df_movie_info = pd.read_table('data/rt.movie_info.tsv')
df_reviews = pd.read_table('data/rt.reviews.tsv',encoding = 'unicode_escape' )
df_movies= pd.read_csv('data/tmdb.movies.csv')
df_movie_budgets = pd.read_csv('data/tn.movie_budgets.csv')



# 2.2 Previewing the Datasets and getting information about dataset

# 2.2.1 df_movie_gross

In [146]:
#previewing df_movie_gross

df_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [78]:
# Accessing more information about df_movie_gross
df_movie_gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This shows that the datas has 3 data types : object, float64 and interger.
  The column names include title, studio, domestic_gross, foreign_gross and year
  

In [101]:
df_movie_gross.shape

(3387, 5)

This shows that the data 3387 rows and 5 columns

In [102]:
df_movie_gross.describe

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

Above are the descriptive statistics of the data

# 2.2.2 Previewing df_movie_info

In [104]:
df_movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [105]:
df_movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


This shows that the datas has 2 data types : object and integer.
  The columns include id, synopsis, rating, director, writer, theater_date, dvd_date, currency, box_office, runtime and studio

In [106]:
df_movie_info.shape

(1560, 12)

The data has 1560 columns and 12 rows

In [124]:
df_movie_info.describe()

id
count  1560.000000
mean   1007.303846
std     579.164527
min       1.000000
25%     504.750000
50%    1007.500000
75%    1503.250000
max    2000.000000

Above are the descriptive statistics of the data

# 2.2.3 Previewing df_reviews


In [125]:
df_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [126]:
df_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


The data has 2 data types : Interger and object. 
The columns include id, review, rating, fresh, critic, top_critic, publisher and date

In [127]:
df_reviews.shape

(54432, 8)

The data has 54432 columns and 8 rows

In [128]:
df_reviews.describe()

id    top_critic
count  54432.000000  54432.000000
mean    1045.706882      0.240594
std      586.657046      0.427448
min        3.000000      0.000000
25%      542.000000      0.000000
50%     1083.000000      0.000000
75%     1541.000000      0.000000
max     2000.000000      1.000000

Above are the descriptive statistics of the data

# 2.2.4 Previewing df_movies

In [129]:
df_movies.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [130]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


The data has 3 types : interger, object and float. 
The columns include genre_ids, id, original_langauge, oriinal_title, popularity, release_date, title, vote_avarage and vote_count

In [131]:
df_movies.shape

(26517, 10)

The data has 26517 rows and 10 columns 

In [132]:
df_movies.describe()

Unnamed: 0             id    popularity  vote_average    vote_count
count  26517.00000   26517.000000  26517.000000  26517.000000  26517.000000
mean   13258.00000  295050.153260      3.130912      5.991281    194.224837
std     7654.94288  153661.615648      4.355229      1.852946    960.961095
min        0.00000      27.000000      0.600000      0.000000      1.000000
25%     6629.00000  157851.000000      0.600000      5.000000      2.000000
50%    13258.00000  309581.000000      1.374000      6.000000      5.000000
75%    19887.00000  419542.000000      3.694000      7.000000     28.000000
max    26516.00000  608444.000000     80.773000     10.000000  22186.000000

Above are the descriptive statistics of the data

# 2.2.5 Previewing df_movie_budgets


In [133]:
df_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [134]:
df_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


The data has 2 types : interger and object.
The columns includes id, rerlease_date, production_budget, domestic_gross and worldwide_gross

In [135]:
df_movie_budgets.shape

(5782, 6)

The data has 5782 columns nad 6 rows

In [136]:
df_movie_budgets.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

Above are the descriptive statistics of the data

# 2.2.6 Previewing df_movie_basics

In [148]:
#reviewing df_movie_basics

df_movie_basics = pd.read_sql_query("""
SELECT *
FROM movie_basics
;
""", conn)
df_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [149]:
df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The data has 3 data types : object, interger and float.
    Columns include movie_id, primary_title, original_title, start_year, runtime_minutes and genres

In [150]:
df_movie_basics.shape

(146144, 6)

The data has 146144 columns and 6 rows

In [151]:
df_movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

Above are the descriptive statistics of the data

# 2.2.7 Previewing df_movie_ratings

In [152]:
#previewing movie_ratings

df_movie_ratings = pd.read_sql_query("""
SELECT *
FROM movie_ratings
;
""", conn)
df_movie_ratings.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [153]:
df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The data has 3 data types : object, interger and float. 
    The columns are movie_id, averagerating and numvotes

In [154]:
df_movie_basics.shape

(146144, 6)

The data has 73856 columns and 3 rows

In [155]:
df_movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

Above are the descriptive statistics of the data

# 2.3 Relevant dataset and the limitation

After carefully scrutinizing the above datasets, 3 datasets stand out and they most relevant for my analysis and are able to answer the 3 business questions
This are 3 datasets<br> **1. df_movie_gross** <br> **2. df_movie_basics**<br> **3. df_movie_ratings**

One  limitation is the movie industry evolves drastically and the data we are using may cannot anticipate the changes that
may occur. A perfect example is that as the years progress, the new generation may be inclined to new types of genres of movies.
Hence by the time Microsoft has already started there movie industry, they could be changes is movie preference

# 2.3 Data Cleaning

# 2.3.1 df_movie_gross

In [71]:
df_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [63]:
# Accessing more information about df_movie_gross

df_movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [64]:
#getting shape

df_movie_gross.shape

(3387, 5)

# 2.3.2 df_movie_basics

In [72]:
df_movie_basics

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [65]:
# Accessing more information about df_movie_basics

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   movie_id         5 non-null      object 
 1   primary_title    5 non-null      object 
 2   original_title   5 non-null      object 
 3   start_year       5 non-null      int64  
 4   runtime_minutes  4 non-null      float64
 5   genres           5 non-null      object 
dtypes: float64(1), int64(1), object(4)
memory usage: 368.0+ bytes


In [67]:
#getting shape

df_movie_basics.shape

(5, 6)

# 2.3.3 df_movie_ratings

In [73]:
df_movie_ratings

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [68]:

# Accessing more information about df_movie_basics

df_movie_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   movie_id       5 non-null      object 
 1   averagerating  5 non-null      float64
 2   numvotes       5 non-null      int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 248.0+ bytes


In [69]:
#getting shape

df_movie_ratings.shape

(5, 3)

# 2.4 Merging Datasets

# 3. Exploratory Data Analysis (EDA)¶